src/codecs/format/json.rs

```diff
@@ -1,12 +1,16 @@
 use std::convert::TryInto;
 
-use bytes::Bytes;
+use bytes::{BufMut, Bytes};
 use chrono::Utc;
 use serde::{Deserialize, Serialize};
 use smallvec::{smallvec, SmallVec};
+use tokio_util::codec::Encoder;
 
 use crate::{
-    codecs::decoding::{BoxedDeserializer, Deserializer, DeserializerConfig},
+    codecs::{
+        decoding::{BoxedDeserializer, Deserializer, DeserializerConfig},
+        encoding::{BoxedSerializer, SerializerConfig},
```

Comment:
Not being particularly familiar with the work so far... why isn't there a matching `encoding::Serializer` to the `decoding::Deserializer`?

Reply:
That's a good question! There is:

https://github.com/vectordotdev/vector/blob/eadf6c8a36296889898b54d6afd05d2cd3b10b6b/src/codecs/encoding.rs#L93-L97

it just happens to have a blanket implementation when we implement `tokio_util::codec::Encoder`:

https://github.com/vectordotdev/vector/blob/eadf6c8a36296889898b54d6afd05d2cd3b10b6b/src/codecs/encoding.rs#L99-L104